File: p22/p22.py
#!/usr/bin/python

def worth(name): return sum(ord(letter) - ord('A') + 1 for letter in name)

txt = open("p022_names.txt")
fl  = txt.read()[1:][:-1].split('","')
fl.sort()
 
answer = sum((i+1) * worth(fl[i]) for i in xrange(0, len(fl)))

print "Answer %s " % answer

File: p22/p22.erl
#!/usr/bin/env escript
%% -*- erlang -*-
%%! -smp enable -sname p22
% vim:syn=erlang

-mode(compile).

worth(W) -> lists:foldl(fun(E,A)-> A + E - hd("A") + 1 end,0,W).

main(_) ->
    {ok, Data} = file:read_file("p022_names.txt"),
    S = size(Data)-2,
    <<"\"", D:S/binary, "\"">> = Data,
    L = lists:sort(binary:split(D,<<"\",\"">>,[global])),
    io:format("Answer: ~p ~n", [sum(L,1,0)]).

sum([], _, A)    -> A;
sum([H|T], I, A) -> sum(T,I+1,A+worth(binary_to_list(H))*I).
